common/src/main/java/apoc/util/Util.java

```diff
@@ -109,6 +109,9 @@ public class Util {
 
     public static final int REDIRECT_LIMIT = 10;
 
+    public static String INVALID_QUERY_MODE_ERROR = "This procedure allows for READ-only, non-schema based queries. " +
+            "It is therefore not possible to perform writes or query the database with commands such as SHOW CONSTRAINTS/INDEXES.";
```

Comment:
Might be worth adding a non-schema write query to each test as well, to make sure that the write part of the error message holds.

Reply:
The writes are actually caught in a different place, but I liked the more generic message which is why I wrote it like this, as it still true you can't write in any of these queries